dotcom-rendering/src/components/YoutubeAtom/YoutubeAtomPlayer.tsx

```diff
@@ -349,21 +344,39 @@ const createImaManagerListeners = (uniqueId: string) => {
 	};
 };
 
+const isSignedIn = async (): Promise<boolean> => {
```

Comment:
This is only used for the ads enabled experience so moved from a custom hook to an promisified function. This enables it to be called conditionally whereas that would not be possible with a hook, or without modifying the hook itself.